### PR TITLE
feature[canny&onesignal]:: Refactoring code.

### PR DESCRIPTION
**Pull request by ItsSudip**
## Description of the change

Refactoring code.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/one_signal/transform.js [hunk 1]
```diff
@@ -157,11 +157,11 @@ const trackResponseBuilder = (message, { Config }) => {
         properties[item.propertyName] ||
```
Comment (koladilip) [inline v0/destinations/one_signal/transform.js:157]: if (typeof properties[item.propertyName] === "string") should cover these if (properties[item.propertyName] || properties[item.propertyName] === "") right?

Comment (shrouti1507) [reply v0/destinations/one_signal/transform.js:157]: yeah, I also think so.

File: v0/destinations/one_signal/transform.js [hunk 2]
```diff
@@ -157,11 +157,11 @@ const trackResponseBuilder = (message, { Config }) => {
...
-        if (eventAsTags && typeof properties[item.propertyName] === "string") {
-          tags[`${event}_${[item.propertyName]}`] =
-            properties[item.propertyName];
-        } else if (typeof properties[item.propertyName] === "string") {
-          tags[item.propertyName] = properties[item.propertyName];
+        if (typeof properties[item.propertyName] === "string") {
+          const tagName = eventAsTags
+            ? `${event}_${[item.propertyName]}`
+            : item.propertyName;
+          tags[tagName] = properties[item.propertyName];
```
Comment (shrouti1507) [inline v0/destinations/one_signal/transform.js:164]: @ItsSudip isn't `eventAsTags` is a boolean value? is it possible to use an object key as a boolean value?

Comment (koladilip) [reply v0/destinations/one_signal/transform.js:164]: const tagName = eventAsTags  ? `${event}_${[item.propertyName]}` : item.propertyName; @shrouti1507 (ternary operator)

Comment (ItsSudip) [reply v0/destinations/one_signal/transform.js:164]: Yeah, @koladilip. @shrouti1507 we are not using eventAsTags directly as some key of object we are using that as deciding variable.

File: v0/destinations/one_signal/transform.js (hunk 1 above)
Comment (ItsSudip) [reply v0/destinations/one_signal/transform.js:157]: Updated.

**Review by koladilip — DISMISSED** [review]

File: v0/destinations/canny/data/identifyConfig.json [hunk 3]
```diff
@@ -34,7 +34,12 @@
   },
   {
     "destKey": "avatarURL",
-    "sourceKeys": ["context.traits.avatarURL", "traits.avatarURL"],
+    "sourceKeys": [
+      "context.traits.avatar",
```
Comment (arnab-p) [inline v0/destinations/canny/data/identifyConfig.json:38]: We give precedence to the top level traits over the context.traits. Let's fix that

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: v0/destinations/canny/data/identifyConfig.json (hunk 3 above)
Comment (ItsSudip) [reply v0/destinations/canny/data/identifyConfig.json]: Updated.

File: v0/destinations/canny/data/identifyConfig.json [hunk 4]
```diff
@@ -34,7 +34,12 @@
   },
   {
     "destKey": "avatarURL",
-    "sourceKeys": ["context.traits.avatarURL", "traits.avatarURL"],
+    "sourceKeys": [
+      "traits.avatar",
+      "traits.avatarURL",
+      "context.traits.avatar",
+      "context.traits.avatarURL"
+    ],
```
Comment (arnab-p) [inline v0/destinations/canny/data/identifyConfig.json:42]:
```suggestion
    "sourceKeys": [
      "traits.avatar",
       "context.traits.avatar",
       "traits.avatarURL",
       "context.traits.avatarURL"
    ],
```

**Review by arnab-p — DISMISSED** [review]

File: v0/destinations/canny/data/identifyConfig.json (hunk 4 above)
Comment (ItsSudip) [reply v0/destinations/canny/data/identifyConfig.json]: Updated

File: v0/destinations/canny/data/identifyConfig.json [hunk 5]
```diff
@@ -34,7 +34,12 @@
   },
   {
     "destKey": "avatarURL",
-    "sourceKeys": ["context.traits.avatarURL", "traits.avatarURL"],
+    "sourceKeys": [
+      "context.traits.avatar",
+      "traits.avatar",
+      "context.traits.avatarURL",
+      "traits.avatarURL"
+    ],
```
[the remark this answers is not in the document]
Comment (ItsSudip) [reply v0/destinations/canny/data/identifyConfig.json]: Added

**Review by arnab-p — APPROVED** [review]

**Review by sandeepdsvs — APPROVED** [review]

**Pull request merged**